### DISCUSSION TITLE
ActionScripts3 not working.

### DISCUSSION THREAD

**Discussion by kamalsaini2787** [ruffle-rs/ruffle#2837] [General]
I am trying to access my internet device with ruffle but is says "**This SWF contains ActionScripts 3 which is not yet supported by ruffle"**
please help !!!!

**Comment by Poopooracoocoo** [comment]
The SWF you're trying to play uses ActionScript 3, the newer language for SWFs, usually from 2006 and onwards. Ruffle does not currently support it. Support for it may take several months. Consider contributing to Ruffle's code or donating ;)

**Reply by imallet** [reply]
> Consider contributing to Ruffle's code or donating ;)

Dear Ruffle community,
I work in a local government in France (Lyon) and we have several GIS web applications which code include a large % of ActionScript 3 code.
We are looking for an opensource solution what would enable us to continue using theses applications for about 2 more years (in order to have time enough to develop new GIS web or mobile applications).
We have read that Ruffle is only 10% compatible with ActionScript 3 code : what could speed up the process of making Ruffle fully compatible with ActionScript 3 ?
My organization would agree to donate to Ruffle community for theses developments, but as a local government we need quotes and planning.
Do you think it's possible ?

Thank you in advance for you answer.
NB : If this issue is not the right one for such question, please tell me where to ask it.

Best regards,

**Reply by flashliberty5467** [reply]
donations help but devolopment wont nessarly go any faster i recommend considering awayFL as thier compatible with AS3 

but kasm cloud browser is avalible for consideration as it runs flash player in the cloud
https://www.kasmweb.com/

puffin browser is also another cloud flash player browser
https://www.puffin.com/

and there is cheerpxflash player https://leaningtech.com/cheerpx-for-flash/

ruffle is not going to get AS3 support for a while you can use the above soulutions while waiting for ruffle

**Reply by DaVince** [reply]
> We have read that Ruffle is only 10% compatible with ActionScript 3 code : what could speed up the process of making Ruffle fully compatible with ActionScript 3 ?
> My organization would agree to donate to Ruffle community for theses developments, but as a local government we need quotes and planning.

A good way would be to hire one or more developers familiar with Rust (and preferably already involved in this project in some way or another), who would then be dedicated to contributing to the project.

Keep in mind that while you do not need to share the source back, it would be very very nice if you could share the source back, if the government agrees to that/allows it. [It's just good practice as you allow many more people to benefit from the work done](https://www.apache.org/foundation/license-faq.html#Must-Contribute).

**Reply by imallet** [reply]
I agree with you. The problem is that the choice of our local government is to buy Harman licences for 2 years, to extend usage of the old Flash applications for 2 more years, in order to have time enough to develop new applications without Flash.

I am very sorry about this decision but it's validated now :-(

**Comment by flashliberty5467** [comment]
i recommend using puffin browser https://www.puffin.com/

or kasm browser https://www.kasmweb.com/

**Reply by imallet** [reply]
Thanks a lot for your answer and suggestions : I'm going to have a look to these solutions.

Best regards

**Reply by parkerlreed** [reply]
If you care about/have any sensitive information DO NOT USE Puffin. The way they "support" flash is rendering the content on their servers (read China) and sending it back to you. Security nightmare.

**Reply by Bale001** [reply]
> If you care about/have any sensitive information DO NOT USE Puffin. The way they "support" flash is rendering the content on their servers (read China) and sending it back to you. Security nightmare.

I've never heard of that. I also would assume that means playing flash games in puffin would be super slow, but I havn't had that issue while using puffin. Do you have more info about this?

**Reply by Bale001** [reply]
> If you care about/have any sensitive information DO NOT USE Puffin. The way they "support" flash is rendering the content on their servers (read China) and sending it back to you. Security nightmare.

Actually nevermind... just did some research and that is crazy

**Reply by parkerlreed** [reply]
> > If you care about/have any sensitive information DO NOT USE Puffin. The way they "support" flash is rendering the content on their servers (read China) and sending it back to you. Security nightmare.
> 
> Actually nevermind... just did some research and that is crazy

Yeah this is the same way Opera used to be so fast on low power devices. They rendered the web pages remotely and sent the tiled images back to the device.

**Reply by azaslavskis** [reply]
Really 😱. Can you drop source please? Thanks for letting me know .

Some lists that I found:


https://www.engadget.com/2020-01-19-opera-accused-of-predatory-loan-apps.html


https://uk.pcmag.com/browsers/131976/chrome-edge-firefox-opera-or-safari-which-browser-is-best?p=1

https://www.mozilla.org/en-GB/firefox/browsers/compare/opera/

https://uk.pcmag.com/browsers/131976/chrome-edge-firefox-opera-or-safari-which-browser-is-best?p=1

https://www.quora.com/What-does-Google-know-about-me

https://www.quora.com/Is-the-Opera-browser-faster-than-Google-Chrome

**Comment by DiHubKi** [comment]
rip

**Comment by nlangdon1** [comment]
kamalsaini2787, I'd give up on Ruffle ever working with AS3.  I was all excited when I first found Ruffle and even donated money, but since I started tracking the "Progress" is hasn't changed from 85% and 15% comparability and they admit that AS3 is basically a pipe dream.

**Reply by n0samu** [reply]
Check it now 😉

**Comment by Bale001** [comment]
> kamalsaini2787, I'd give up on Ruffle ever working with AS3.  I was all excited when I first found Ruffle and even donated money, but since I started tracking the "Progress" is hasn't changed from 85% and 15% comparability and they admit that AS3 is basically a pipe dream. 

The progress bar is not accurate. You can look at the pull requests and search for pull requests starting with avm2 to see all the new avm2 updates

**Reply by DaVince** [reply]
Additionally, you can keep an eye on the [commits](https://github.com/ruffle-rs/ruffle/search?o=desc&q=avm2&s=committer-date&type=commits).

**Comment by nlangdon1** [comment]
Glad to hear.


Sent from my Samsung Note 10+


From: EmperorBale ***@***.***>Sent: Monday, July 12, 2021, 11:26 AMTo: ruffle-rs/ruffleCc: nlangdon1; CommentSubject: Re: [ruffle-rs/ruffle] ActionScripts3 not working. (#2837)

kamalsaini2787, I'd give up on Ruffle ever working with AS3.  I was all excited when I first found Ruffle and even donated money, but since I started tracking the "Progress" is hasn't changed from 85% and 15% comparability and they admit that AS3 is basically a pipe dream.

The progress bar is not accurate. You can look at the pull requests and search for pull requests starting with avm2 to see all the new avm2 updates

**Comment by nlangdon1** [comment]
Yes, it really sucks, Try Puffin Browser for full Flash support.
Sent from my Samsung Note 10+From: Alex Zaslavskis ***@***.***>Sent: Thursday, September 2, 2021, 3:45 PMTo: ruffle-rs/ruffleCc: nlangdon1; CommentSubject: Re: [ruffle-rs/ruffle] ActionScripts3 not working. (#2837)

Really 😱.I a going to remove it from my pc rn . Thanks for letting me know .

**Comment by nlangdon1** [comment]
Really, it works?  Sent from my T-Mobile 4G LTE DeviceGet Outlook for Android

**Comment by nlangdon1** [comment]
Any hints on where to find the Firefox plugin and hot to install it?

**Reply by n0samu** [reply]
https://addons.mozilla.org/en-US/firefox/addon/ruffle_rs/

**Comment by nlangdon1** [comment]
I will try that link, but after searching around I did find a link and still got:

 



 

Just in case the link you sent is updated I’ll check it out.

 

Thanks,

 

-Nick

**Comment by nlangdon1** [comment]
It was the same link.   L   Pale Moon browser does allow the Broadway to stream though, but it would be nice if it worked on any browser (and my Android!).

**Comment by nlangdon1** [comment]
How do I get around the browser not allowing me to install the browser extension?

**Reply by n0samu** [reply]
For Android, we recommend using the [Kiwi Browser](https://kiwibrowser.com/) along with the Ruffle Chrome extension.

**Comment by nlangdon1** [comment]
My Broadway device already works with android I'm trying to get at working on my computer.Sent from my T-Mobile 4G LTE DeviceGet Outlook for Android

**Reply by n0samu** [reply]
You can install the Ruffle extension on your computer's Firefox or Chrome browser:
- Chrome: https://chrome.google.com/webstore/detail/ruffle/donbcfbmhbcapadipfkeojnmajbakjdc
- Firefox: https://addons.mozilla.org/firefox/addon/ruffle_rs
